### PR TITLE
Fix jemalloc compliation problem on macOS

### PR DESCRIPTION
**Pull request by adamretter**
Closes https://github.com/facebook/rocksdb/issues/7269

I have only tested this on macOS, let's see what CI makes of it for the other platforms...

### CODE REVIEW

**Comment by pdillinger** [comment]
Hole left by #6613? Can we pick one syntax to unify on or is there a distinction?

**Comment by adamretter** [comment]
@pdillinger I have now updated this to use consistent syntax and follow the jemalloc declarations more closely. I *think* this should be okay, but let's see what CI says...

File: port/jemalloc_helper.h [hunk 1]
```diff
@@ -40,23 +40,34 @@ static inline bool HasJemalloc() { return true; }
...
-extern "C" int mallctlnametomib(const char*, size_t*, size_t*)
-    __attribute__((__nothrow__, __weak__));
-extern "C" int mallctlbymib(const size_t*, size_t, void*, size_t*, void*,
-                            size_t) __attribute__((__nothrow__, __weak__));
-extern "C" void malloc_stats_print(void (*)(void*, const char*), void*,
-                                   const char*) __attribute__((__nothrow__, __weak__));
-extern "C" size_t malloc_usable_size(JEMALLOC_USABLE_SIZE_CONST void*)
-    JEMALLOC_CXX_THROW __attribute__((__weak__));
+extern "C" JEMALLOC_ALLOCATOR JEMALLOC_RESTRICT_RETURN void JEMALLOC_NOTHROW *
+je_mallocx(size_t, int) JEMALLOC_ATTR(malloc) JEMALLOC_ALLOC_SIZE(1)
```
Comment (pdillinger) [inline port/jemalloc_helper.h:44]: Is there a reason this changed from `mallocx` to `je_mallocx`? `mallocx` is still referenced below in `HasJemalloc()`

Comment (adamretter) [reply port/jemalloc_helper.h:44]: @pdillinger Sorry, that was unintentional. Should be fixed now.

**Pull request closed without merging**